contracts/consensus/Consensus.sol

```diff
@@ -505,63 +515,67 @@ contract Consensus is MasterCopyNonUpgradable, CoreLifetimeEnum, MosaicVersion,
     }
 
     /**
-     * @notice Validator joins the core, when core lifetime status is
-     *         is active. This is called by validator address.
+     * @notice Validator joins the core, when core lifetime status is genesis
+     *         or active. This is called by validator address.
      *
      * @dev Function requires:
-     *          - core status should be opened or precommitted.
+     *          - core address is 0.
+     *          - core lifetime status must be genesis or active.
      *
      * @param _metachainId Metachain id that validator wants to join.
-     * @param _core Core address that validator wants to join.
      * @param _withdrawalAddress A withdrawal address of newly joined validator.
      */
     function join(
         bytes32 _metachainId,
-        address _core,
         address _withdrawalAddress
     )
         external
     {
-        // Validate the join params.
-        validateJoinParams(_metachainId, _core, _withdrawalAddress);
+        address core = assignments[_metachainId];
+        require(
+            core != address(0),
+            "Core address is 0."
```

Comment:
Core doesn't exist for given metachain ?

Reply:
Done!